### ISSUE TITLE
bluetooth not availible

### ISSUE THREAD

**Issue by m9ac** [dokterdok/Continuity-Activation-Tool#380]
ive been all over the forum and read loads of others and seen youtube videos on the ins and outs of the process.

I keep getting the error that bluetooth is not available, though Ive installed the new card and run the CAT software.

is it even possible that the card i installed could work fine for wifi but not have any bluetooth functionality whatsoever? I cant find bluetooth in system profiler, using CAT, in my VM running ubuntu. nothing!

I suspect i will need to replace the card once more, but I wondered if anyone had any input before wasting money once again.
 Thanks

**Comment by christianboyle** [comment]
Sounds similar to #361. I'm about to give up and create a new clone from my original system today.

**Comment by m9ac** [comment]
sounds awesome. please let me know when you get it ready for download

**Comment by christianboyle** [comment]
TLDR: This tool isn't necessary if you install things prior to the system initializing them. You will likely need to reinstall/rebuild your system for it to regain BT functionality.

I never ended up finding a fix for this, so I ended up recloning my SSD, but there were a few interesting things:

- I use Broadcom [BCM94360CD](https://www.amazon.com/dp/B00PBSFEHA/) with [PCI-E adapter](https://www.amazon.com/PCI-Wireless-Broadcom-BCM94360CD-BCM94331CD/dp/B00THN281U) and [TP-Link TL-WDN4800 (N900)](https://www.amazon.com/TP-Link-Wireless-Express-Adapter-TL-WDN4800/dp/B007GMPZ0A)
- I installed these cards at different times (BCM first, then WDN a few weeks later)
- Both cards worked when installed without needing additional software, but Handoff and Continuity options were never available after installing WDN (I assumed the presence of the card would turn on the missing features and they'd show in Preferences > General, but they didn't)
- I ran CAT and it broke whatever the system uses to see the Bluetooth firmware, resulting in permanently broken Bluetooth (I tried reseating both cards and it had no effect)
- Gave up and re-cloned by iMac SSD with CCC and got back up and running with my old Clover config

Funny thing is, after this new install everything works and Handoff/Continuity are now functional. My guess is that something in MacOS expected both BT and WiFi to be present and when they weren't it never had a way to recover and running CAT merely confirmed this low-level brokenness.

**Comment by sparky672** [comment]
I briefly had this issue on Sierra 10.12.3 after installing my new card on a Mac Pro 5,1  Coincidentally, I could not hook up the Bluetooth antenna wire, so I figured it had to do with that.   I simply used my Bluetooth dongle for a few days while I waited for my antenna extension cable.  Sometime while waiting for my part to arrive, I removed the dongle and Bluetooth was suddenly activated.   Bluetooth was showing as available to the system, although the signal was too weak for it to be picked up by any devices.  Once the antenna cable was installed, Bluetooth went back to working normally.

It took a bit more effort to get Continuity fully functional, but eventually got that sorted out too.  I tried a whole lot of things, but believe the most critical part was trashing Bluetooth prefs, zapping the PRAM, and logging out of iCloud and rebooting before logging back into iCloud.

**Comment by patriot1889** [comment]
I had this happen today after I forced the CAT to patch my system.
I fixed it by booting into recovery and reinstalling macOS over the current install.

Everything was still in place when I rebooting and the bluetooth had come back to life.

Furthermore, doing this helped me get handoff working. I logged out of iCloud before reinstalling and then when I rebooted, ran CAT, rebooted and logged into iCloud. Left it overnight and now everything works fine.

iMac 2011 with hardware mod. My Mac only needed a plist file patching, so it works with SIP enabled :)

**Comment by iulianstar** [comment]
Hello all,

I am on OS Yosemite using a MacBook Pro early 2011, with SSD. Read a post about upgrading to Sierra and better upgrade de WIFI card and use CAT so to benefit for the new tools.
I have bought this on Ali: https://www.aliexpress.com/item/3Pcs-Original-Bluetooth-4-0-wifi-card-Airport-Card-for-Macbook-Pro-A1278-A1286-2011-2012year/32657027888.html?spm=2114.13010608.0.0.j52hxz. 
Can someone help me with a step by step upgrade from Yosemite to Sierra after I will replace the card?
